substrate/frame/message-queue/src/lib.rs

```diff
@@ -516,18 +537,51 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// Message discarded due to an error in the `MessageProcessor` (usually a format error).
-		ProcessingFailed { id: [u8; 32], origin: MessageOriginOf<T>, error: ProcessMessageError },
+		ProcessingFailed {
+			/// The `blake2_256` hash of the message.
+			id: [u8; 32],
```

Comment:
Is it better to use the `H256` type here?

Reply:
I guess we could change it, but its a breaking change and so far this MR has not been...